packages/nextjs/src/components-nextjs/StorySeo/StorySeo.tsx

```diff
@@ -43,8 +43,8 @@ export function StorySeo({ story, noindex }: Props) {
                 title={seoTitle}
                 description={seoDescription}
                 canonical={canonical}
-                noindex={noindex ?? !indexable}
-                nofollow={noindex ?? !indexable}
+                noindex={noindex && !indexable}
+                nofollow={noindex && !indexable}
```

Comment:
After thinking more, I believe it should be like this:

```tsx
noindex={noindex || !indexable}
nofollow={noindex || !indexable}
```

Same change for `PageSeo.tsx`

Reply:
Was about to post the same comment 😃 
With current code, the seo setting will not take effect if the site is indexable

Reply:
Inverted boolean flags are tricky! :)